### PR TITLE
Make the fake command factory return the clientset with appropriate rest clients for all the API groups.

### PR DESCRIPTION
**Pull request by madhusudancs**
Please review only the last commit here. This is based on PR #35865 which will be reviewed independently.

Design Doc: PR #34484

cc @kubernetes/sig-cluster-federation @nikhiljindal

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35866)

### CODE REVIEW

File: pkg/kubectl/cmd/testing/fake.go [hunk 1]
```diff
@@ -405,14 +405,22 @@ func (f *fakeAPIFactory) JSONEncoder() runtime.Encoder {
...
+	clientset.AuthenticationClient.RESTClient.Client = fakeClient.Client
+	clientset.AuthorizationClient.RESTClient.Client = fakeClient.Client
+	clientset.AutoscalingClient.RESTClient.Client = fakeClient.Client
+	clientset.BatchClient.RESTClient.Client = fakeClient.Client
+	clientset.CertificatesClient.RESTClient.Client = fakeClient.Client
+	clientset.ExtensionsClient.RESTClient.Client = fakeClient.Client
+	clientset.RbacClient.RESTClient.Client = fakeClient.Client
+	clientset.StorageClient.RESTClient.Client = fakeClient.Client
+	clientset.AppsClient.RESTClient.Client = fakeClient.Client
+	clientset.PolicyClient.RESTClient.Client = fakeClient.Client
```
Comment (janetkuo) [inline pkg/kubectl/cmd/testing/fake.go]: `DiscoveryClient` wasn't covered. Is it intentional?

Comment (janetkuo) [inline pkg/kubectl/cmd/testing/fake.go]: If one adds more clients in `Clientset`, how do we make sure one swap the fake REST client here?

Comment (madhusudancs) [inline pkg/kubectl/cmd/testing/fake.go]: @janetkuo 

> DiscoveryClient wasn't covered. Is it intentional?

Oops. That wasn't intentional. Copy-paste fail. Added it now.

> If one adds more clients in Clientset, how do we make sure one swap the fake REST client here?

Unfortunately, yes.

Comment (janetkuo) [inline pkg/kubectl/cmd/testing/fake.go]: I think you'll need to rebase because RESTClient isn't there anymore :'(

Comment (janetkuo) [inline pkg/kubectl/cmd/testing/fake.go]: This LGTM then.

Comment (madhusudancs) [inline pkg/kubectl/cmd/testing/fake.go]: @janetkuo thanks!

**Comment by madhusudancs** [comment]
Rebased the PR and squashed the commits. Adding LGTM label.

**Pull request merged**